jax/_src/scipy/signal.py

```diff
@@ -147,11 +147,29 @@ def convolve(in1: Array, in2: Array, mode: str = 'full', method: str = 'auto',
 @implements(osp_signal.convolve2d)
 def convolve2d(in1: Array, in2: Array, mode: str = 'full', boundary: str = 'fill',
```

Comment:
It looks like you're working off an older main branch (these APIs don't use `@implements` any more) I'd suggest syncing now to avoid potentially bigger conflicts later.

Reply:
I could not resolve the conflicts after syncing to the latest main branch. Hence closed this issue and reopened a new PR #21235. Sorry for the inconvenience. 

In PR #21235, I have modified `_convolve_nd` function so that `correlate2d` could also make of use of it.